### ISSUE TITLE
Angular version

### ISSUE THREAD

**Issue by Nephidream** [g00fy-/angular-datepicker#196]
Using angular 1.4.7, when I try to install date-picker through bower, I get
> bower angular-datepicker#*      cached git://github.com/g00fy-/angular-datepicker.git#2.0.0
> bower angular-datepicker#*    validate 2.0.0 against git://github.com/g00fy-/angular-datepicker.git#*
> bower angular#1.2.14            cached git://github.com/angular/bower-angular.git#1.2.14
> bower angular#1.2.14          validate 1.2.14 against git://github.com/angular/bower-angular.git#1.2.14
> bower angular#>=1.3.0           cached git://github.com/angular/bower-angular.git#1.4.7
> bower angular#>=1.3.0         validate 1.4.7 against git://github.com/angular/bower-angular.git#>=1.3.0
> bower angular#^1.4.7 <2         cached git://github.com/angular/bower-angular.git#1.4.7
> bower angular#^1.4.7 <2       validate 1.4.7 against git://github.com/angular/bower-angular.git#^1.4.7 <2
> bower angular#1.4.7             cached git://github.com/angular/bower-angular.git#1.4.7
> bower angular#1.4.7           validate 1.4.7 against git://github.com/angular/bower-angular.git#1.4.7
> bower angular#>=1.2.16 1.4.x    cached git://github.com/angular/bower-angular.git#1.4.7
> bower angular#>=1.2.16 1.4.x  validate 1.4.7 against git://github.com/angular/bower-angular.git#>=1.2.16 1.4.x
> bower angular#>=1.2.18          cached git://github.com/angular/bower-angular.git#1.4.7
> bower angular#>=1.2.18        validate 1.4.7 against git://github.com/angular/bower-angular.git#>=1.2.18
> bower                        ECONFLICT Unable to find suitable version for angular

In the bower.json, the dependencies are
> "dependencies": {
>    "angular": "1.2.14",
>    "moment": "~2.10.6",
>    "moment-timezone": "~0.4.1"
>  },

Shouldn't it be "angular": ">=1.2.14" since it is supposed to work with angular 1.2+ ?

**Comment by DanTalash** [comment]
I think the angular version needs to be at least 1.3, as the library is using some features of angular which were added in 1.3

* $validators are used [here](https://github.com/g00fy-/angular-datepicker/blob/master/app/scripts/input.js#L95) and [here](https://github.com/g00fy-/angular-datepicker/blob/master/app/scripts/input.js#L104). 
  * I cannot find any official documentation that explicitly states what version of angular the $validators pipeline was added, but many blog posts mention that it is [new in 1.3](http://busypeoples.github.io/post/forms-in-angularjs/).
  * The [docs for 1.2](https://code.angularjs.org/1.2.27/docs/api/ng/type/ngModel.NgModelController) make no mention of $validators.
  * The [docs for 1.3](https://code.angularjs.org/1.3.19/docs/api/ng/type/ngModel.NgModelController) have information on $validators.
* $watchGroup is used in [this PR](https://github.com/g00fy-/angular-datepicker/pull/198) in the [date-range directive](https://github.com/DanTalash/angular-datepicker/commit/f38616195ee1561c7bc789321a80cbec6b674b60#diff-2bfa03fd33c6f3edf8a26ca8a3bd92aeR64). Again, this is a feature of angularjs 1.3+
  * The [docs for 1.2](https://code.angularjs.org/1.2.27/docs/api/ng/type/$rootScope.Scope) makes no mention of $watchGroup.
  * The [docs for 1.3](https://code.angularjs.org/1.3.19/docs/api/ng/type/$rootScope.Scope) have information on $watchGroup.

If support for angular 1.2 is required, $validators and $watchGroup cannot be used, and must be replaced with something else.

**Comment by Nephidream** [comment]
Yes, but I use Angular 1.4.7, so there should not be compatibility issues with date-picker which requires 1.2+ no ? (the bower.json that I mentionned is [the one of date-picker](https://github.com/g00fy-/angular-datepicker/blob/master/bower.json), sorry if I was not clear enough)

**Comment by alexeyzimarev** [comment]
I have the same issue. Current bower.json has strict dependency on angular 1.2.14 and this does not work with 1.4.7.

**Comment by valera33** [comment]
same issue

**Comment by timmyomahony** [comment]
also the same issue

**Comment by mcranston18** [comment]
+1

**Comment by steve-mcqs** [comment]
+1

**Comment by dejan7** [comment]
+1

**Comment by randyv12** [comment]
+1

**Comment by AdamHess** [comment]
I'm a little mad that this hasnt been fixed since it's so easy to there are probably a dozen pull requests for it. 

Here's the simple fix for your environment: 

add the following to your bower.json 

  "resolutions": {
    "angular": "~1.4.7"   <-- or whatever version of angular you are using
  }


this will force bower to select the version of angular you want instead of being stuck with the 1.2 version the library uses.

**Comment by larryspace** [comment]
+1
